mr/src/main/java/org/apache/iceberg/mr/mapred/IcebergSerDe.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.mapred;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Properties;
+import javax.annotation.Nullable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.serde2.AbstractSerDe;
+import org.apache.hadoop.hive.serde2.SerDeException;
+import org.apache.hadoop.hive.serde2.SerDeStats;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.io.Writable;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.mr.mapred.serde.objectinspector.IcebergObjectInspector;
+
+public class IcebergSerDe extends AbstractSerDe {
+
+  private ObjectInspector inspector;
+
+  @Override
+  public void initialize(@Nullable Configuration configuration, Properties serDeProperties) throws SerDeException {
+    final Table table;
+
+    try {
+      table = TableResolver.resolveTableFromConfiguration(configuration, serDeProperties);
+    } catch (IOException e) {
+      throw new UncheckedIOException("Unable to resolve table from configuration: ", e);
```

Comment:
I didn't realize Java added an `UncheckedIOException` in 8. We have one that is `RuntimeIOException`. We should probably convert Iceberg over to using the standard Java one.